code-challenges.test.js

```diff
@@ -15,42 +16,137 @@
+        expect(codedMes(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+//  ● codedMes › a function that takes in a string and returns a coded message that converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0
+
+// ReferenceError: codedMes is not defined
 
 
 // b) Create the function that makes the test pass.
 
+// Create a function named codedMes with a string as the input parameter
+// Create a new variable and assign it to a built in method to the string to replace 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0
+// Return the new variable
+
+
+const codedMes = (string) => {
+let codeString = string.replace(/a/g, 4).replace(/e/g, 3).replace(/i/g, 1).replace(/o/g, 0).replace(/A/g, 4).replace(/E/g, 3).replace(/I/g, 1).replace(/O/g, 0)
+return codeString
+}
+
 
 // --------------------2) Create a function that takes in an array of words and a single letter and returns all the words that contain that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
-const letterA = "a"
-// Expected output: ["Apple", "Banana", "Orange"]
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+describe("findLetter", () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"]
+    const letterA = "a"
+    // Expected output: ["Apple", "Banana", "Orange"]
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
+    const letterE = "e"
+    // Expected output: ["Cherry", "Blueberry", "Peach"]
+    it("a function that takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+        expect(findLetter(arrayOfWords1,letterA)).toEqual(["Apple", "Banana", "Orange"])
+        expect(findLetter(arrayOfWords2,letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+    })
+})
+
+//  ● findLetter › a function that takes in an array of words and a single letter and returns all the words that contain that particular letter
+
+// ReferenceError: findLetter is not defined
 
 
 // b) Create the function that makes the test pass.
 
+// Create a function named findLetter that takes two inputs as a parameter: first one array and second parameter: single letter (string)
+// Create an empty array named filArray
+// Iterate over each value in the input array to find if the input letter/string is present
+    // If present, push the value into finArray
+// Return finArray
 
-// --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
+const findLetter = (array,letter) => {
+    let filArray = []
+   array.map(value => {
+    if (value)array.includes(letter)
+    filArray = filArray.push()
```

Comment:
Just some things to consider

1. Why are reassigning the variable? Maybe consider using a different name for the variable that is storing the push method.
2. What are you pushing to you your filArray?
3. Have you considered uppercase or lower case letter?